flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/RecordWriterOutput.java

```diff
@@ -48,20 +47,19 @@
 
     private SerializationDelegate<StreamElement> serializationDelegate;
 
-    private final StreamStatusProvider streamStatusProvider;
-
     private final boolean supportsUnalignedCheckpoints;
 
     private final OutputTag outputTag;
 
     private final WatermarkGauge watermarkGauge = new WatermarkGauge();
 
+    private StreamStatus announcedStatus = StreamStatus.ACTIVE;
```

Comment:
Is active the default throughout whole Flink?

Reply:
I'd say yes. We start Flink cluster with all channels active, and we wait for Watermarks from all inputs.